### PR TITLE
Add filesystem plugin

### PR DESCRIPTION
**Pull request by josbeir**
Added "yet another" filesystem plugin :-)

=> https://github.com/josbeir/cakephp-filesystem

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -165,6 +165,7 @@ Additional lists you might find useful:
 *Plugins for file manipulation.*
 
 - [FileStorage plugin](https://github.com/burzum/cakephp-file-storage) - Abstract file storage and upload plugin.
+- [Filesystem plugin](https://github.com/josbeir/cakephp-filesystem) - Abstract [Flysystem](http://flysystem.thephpleague.com/) + file entity based upload plugin.
```
Comment (dereuromark) [inline README.md]: Isnt the name technically `Josbeir/Filesystem` ?

Comment (josbeir) [reply README.md]: Technically yes but its a bit weird in representation? Are you proposing something? ;-)

Comment (dereuromark) [reply README.md]: Well, that's the name and thus the standard display name we are using to not confuse people.

Comment (josbeir) [reply README.md]: Fixed !

File: README.md [hunk 2]
```diff
@@ -165,7 +165,7 @@ Additional lists you might find useful:
 *Plugins for file manipulation.*
 
 - [FileStorage plugin](https://github.com/burzum/cakephp-file-storage) - Abstract file storage and upload plugin.
-- [Filesystem plugin](https://github.com/josbeir/cakephp-filesystem) - Abstract [Flysystem](http://flysystem.thephpleague.com/) + file entity based upload plugin.
+- [Josbeir/Filesystem plugin](https://github.com/josbeir/cakephp-filesystem) - Abstract [Flysystem](http://flysystem.thephpleague.com/) + file entity based upload plugin.
```
Comment (dereuromark) [inline README.md]: Alph. order please.

Comment (josbeir) [reply README.md]: Fixed! Did a rebase, you're too fast :-)

**Comment by josbeir** [comment]
Is there an 'evalution' period on these pr's ? :-)

**Comment by dereuromark** [comment]
I think people were just hesitant because there are so many file handling ones already in the list.
Are there any key differences/improvements here?

I do like the docs verbosity :)

**Comment by josbeir** [comment]
I can't speak of the other plugins but the 'cool' thing here is that instead of working with files directly the requirement is that we work with a file entity interface and a simple file entity class (nothing related to the ORM). There is no database requirement by default for storing and manipulating files.

But the beauty of this is that storing entities in the db can be easily done by just swapping the entity class configuration ``entityClass`` with an ORM one, this entity then needs to implement the `FileEntityInterface` interface and add a required method...

And of course we have events, php 7.1 strictness, working with Zend PSR streams, paths and $_FILES..., filename formatters, ...

**Comment by dereuromark** [comment]
Well, I dont think it hurts to add if it is well documented and tested.
It does sound like a valuable addition.

**Comment by josbeir** [comment]
I would think it passes those two requirements :-)

**Comment by ADmad** [comment]
Why does the readme still say "(WIP)"? :)

**Comment by josbeir** [comment]
@ADmad yeah, because it is still WIP, it is not battle tested in production as i'm still working on the project where it is made for. But it works pretty well :-)

I'll remove it because it will probably confuse people...

**Comment by ADmad** [comment]
It can deter potential users, so better to remove. Also better to use `0.x` versioning until you are satisfied with your product :)

**Comment by ADmad** [comment]
@dereuromark Good to merge?

**Comment by dereuromark** [comment]
Where did the plugin load part go?
And composer require josbeir/cakephp-filesystem:dev-master should be only `composer require josbeir/cakephp-filesystem` when you have tags.

**Comment by josbeir** [comment]
Correct, upated.

There is currently no need to load the plugin as there are no components, behaviors, routes, ...

**Pull request merged**